packages/interfaces/src/misc.ts

```diff
@@ -11,3 +12,23 @@ export type PubsubTopic = string;
 export type ContentTopic = string;
 
 export type PeerIdStr = string;
+
+// SK = success key name
+// SV = success value type
+// EK = error key name (default: "error")
+// EV = error value type (default: ProtocolError)
+export type ThisOrThat<
+  SK extends string,
+  SV,
+  EK extends string = "error",
+  EV = ProtocolError
+> =
+  | ({ [key in SK]: SV } & { [key in EK]: null })
+  | ({ [key in SK]: null } & { [key in EK]: EV });
+
+export type ThisAndThat<
```

Comment:
My comment on `SDKProtocolResult` above likely apply to anything that will implement this type.